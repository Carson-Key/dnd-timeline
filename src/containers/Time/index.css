.line {
  border-bottom: 6px solid;
}
