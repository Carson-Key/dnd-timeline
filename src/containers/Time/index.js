import React, { Component } from 'react'
import './index.css'
import { Media } from 'reactstrap'

class Time extends Component {
  render() {
    const { timeline } = this.props
    const events = Object.keys(timeline)
    return events.map((event, i) => {
      return (
        <center key={i}>
          <Media>
            <Media body>
              <Media heading>
                {timeline[event].date}
              </Media>
              {timeline[event].description}
            </Media>
          </Media>
          <br/>
          <div className="line"></div>
          <br/>
        </center>
      )
    })
  }
}

export default Time
