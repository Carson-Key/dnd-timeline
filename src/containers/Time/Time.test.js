import React from "react"
import Time from "./index.js"
import { render } from 'enzyme'
import toJson from "enzyme-to-json";

describe('Render Time ->', () => {

  beforeEach(() => {
    const originalError = console.error
    console.error = jest.fn()
  })

  test('with blank timeline prop', () => {
    const time = render(
      <Time timeline={{}}/>,
    )

    expect(toJson(time)).toMatchSnapshot()
  })

  test('with out year suffix timeline prop', () => {
    const time = render(
      <Time timeline={{
        "1776": [
          {
            description: "America!!!",
          }
        ]
      }}/>,
    )

    expect(toJson(time)).toMatchSnapshot()
  })

  test('with events and year suffix timeline prop', () => {
    const time = render(
      <Time
        timeline={{
          "1776": [
            {
              description: "America!!!",
            }
          ]
        }}
        lessThanYearZero="BCE"
        greaterThanYearZero="AD"
      />,
    )

    expect(toJson(time)).toMatchSnapshot()
  })

  test('with multiple years', () => {
    const time = render(
      <Time
        timeline={{
          "1776": [
            {
              description: "America!!!",
            }
          ],
          "1777": [
            {
              description: "More America!!!",
            }
          ]
        }}
        lessThanYearZero="BCE"
        greaterThanYearZero="AD"
      />,
    )

    expect(toJson(time)).toMatchSnapshot()
  })

  test('with multiple events', () => {
    const time = render(
      <Time
        timeline={{
          "1776": [
            {
              description: "America!!!",
            },
            {
              description: "America 2.0!!!",
            }
          ]
        }}
        lessThanYearZero="BCE"
        greaterThanYearZero="AD"
      />,
    )

    expect(toJson(time)).toMatchSnapshot()
  })

  test('with multiple events and years', () => {
    const time = render(
      <Time
        timeline={{
          "1776": [
            {
              description: "America!!!",
            },
            {
              description: "America 2.0!!!",
            }
          ],
          "1777": [
            {
              description: "More America!!!",
            },
            {
              description: "More America 2.0!!!",
            }
          ]
        }}
        lessThanYearZero="BCE"
        greaterThanYearZero="AD"
      />,
    )

    expect(toJson(time)).toMatchSnapshot()
  })

})
