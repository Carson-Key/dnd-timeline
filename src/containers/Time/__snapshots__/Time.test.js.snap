// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Time -> with blank timeline prop 1`] = `<br />`;

exports[`Render Time -> with events and year suffix timeline prop 1`] = `
Array [
  <br />,
  <center>
    <div
      class="media"
    >
      <div
        class="media-body"
      >
        <h4
          class="media-heading"
        >
          1776 AD
        </h4>
        America!!!
      </div>
    </div>
    <br />
    <div
      class="line"
    />
    <br />
  </center>,
]
`;

exports[`Render Time -> with multiple events 1`] = `
Array [
  <br />,
  <center>
    <div
      class="media"
    >
      <div
        class="media-body"
      >
        <h4
          class="media-heading"
        >
          1776 AD
        </h4>
        America!!!
      </div>
    </div>
    <br />
    <div
      class="line"
    />
    <br />
  </center>,
  <center>
    <div
      class="media"
    >
      <div
        class="media-body"
      >
        <h4
          class="media-heading"
        >
          1776 AD
        </h4>
        America 2.0!!!
      </div>
    </div>
    <br />
    <div
      class="line"
    />
    <br />
  </center>,
]
`;

exports[`Render Time -> with multiple events and years 1`] = `
Array [
  <br />,
  <center>
    <div
      class="media"
    >
      <div
        class="media-body"
      >
        <h4
          class="media-heading"
        >
          1776 AD
        </h4>
        America!!!
      </div>
    </div>
    <br />
    <div
      class="line"
    />
    <br />
  </center>,
  <center>
    <div
      class="media"
    >
      <div
        class="media-body"
      >
        <h4
          class="media-heading"
        >
          1776 AD
        </h4>
        America 2.0!!!
      </div>
    </div>
    <br />
    <div
      class="line"
    />
    <br />
  </center>,
  <center>
    <div
      class="media"
    >
      <div
        class="media-body"
      >
        <h4
          class="media-heading"
        >
          1777 AD
        </h4>
        More America!!!
      </div>
    </div>
    <br />
    <div
      class="line"
    />
    <br />
  </center>,
  <center>
    <div
      class="media"
    >
      <div
        class="media-body"
      >
        <h4
          class="media-heading"
        >
          1777 AD
        </h4>
        More America 2.0!!!
      </div>
    </div>
    <br />
    <div
      class="line"
    />
    <br />
  </center>,
]
`;

exports[`Render Time -> with multiple years 1`] = `
Array [
  <br />,
  <center>
    <div
      class="media"
    >
      <div
        class="media-body"
      >
        <h4
          class="media-heading"
        >
          1776 AD
        </h4>
        America!!!
      </div>
    </div>
    <br />
    <div
      class="line"
    />
    <br />
  </center>,
  <center>
    <div
      class="media"
    >
      <div
        class="media-body"
      >
        <h4
          class="media-heading"
        >
          1777 AD
        </h4>
        More America!!!
      </div>
    </div>
    <br />
    <div
      class="line"
    />
    <br />
  </center>,
]
`;

exports[`Render Time -> with out year suffix timeline prop 1`] = `
Array [
  <br />,
  <center>
    <div
      class="media"
    >
      <div
        class="media-body"
      >
        <h4
          class="media-heading"
        >
          Error!
        </h4>
        America!!!
      </div>
    </div>
    <br />
    <div
      class="line"
    />
    <br />
  </center>,
]
`;
